src/xenia/kernel/user_module.cc

```diff
@@ -804,5 +808,39 @@ void UserModule::Dump() {
   xe::logging::AppendLogLine(xe::LogLevel::Info, 'i', sb.to_string_view());
 }
 
+void UserModule::CalculateHash() {
+  const uint32_t page_size =
+      xex_module()->base_address() < 0x90000000 ? 64 * 1024 : 4 * 1024;
+
+  auto security_info = xex_module()->xex_security_info();
+
+  auto find_code_section_page = [&security_info](bool from_bottom) {
+    for (uint32_t i = 0; i < security_info->page_descriptor_count; i++) {
+      const uint32_t page_index =
+          from_bottom ? i : security_info->page_descriptor_count - i;
+      xex2_page_descriptor page_descriptor;
+      page_descriptor.value =
+          xe::byte_swap(security_info->page_descriptors[page_index].value);
+      if (page_descriptor.info != XEX_SECTION_CODE) {
+        continue;
+      }
+      return page_index;
+    }
+    return (uint32_t)0;
```

Comment:
The cast is not needed here. However, is 0 an invalid index (unlike something like `UINT32_MAX`) actually?